internal/k8s/upstreams/upstreams_test.go

```diff
@@ -0,0 +1,397 @@
+	if probe == nil || probe.PeriodSeconds != 42 {
+		t.Errorf("ServicePort.TargetPort as string failed: %+v", probe)
+	}
+
+	svcPort = v1.ServicePort{
+		TargetPort: intstr.FromInt(80),
+		Protocol:   v1.ProtocolTCP,
+	}
+	probe = findProbeForPods(pods, &svcPort)
+	if probe == nil || probe.PeriodSeconds != 42 {
+		t.Errorf("ServicePort.TargetPort as int failed: %+v", probe)
+	}
+
+	svcPort = v1.ServicePort{
+		Port: 80,
+	}
+	probe = findProbeForPods(pods, &svcPort)
+	if probe == nil || probe.PeriodSeconds != 42 {
+		t.Errorf("ServicePort.Port should match if TargetPort is not set: %+v", probe)
+	}
+
+	svcPort = v1.ServicePort{
+		TargetPort: intstr.FromString("wrong_name"),
+	}
+	probe = findProbeForPods(pods, &svcPort)
+	if probe != nil {
+		t.Errorf("ServicePort.TargetPort should not have matched string: %+v", probe)
+	}
+
+	svcPort = v1.ServicePort{
+		TargetPort: intstr.FromInt(22),
+	}
+	probe = findProbeForPods(pods, &svcPort)
+	if probe != nil {
+		t.Errorf("ServicePort.TargetPort should not have matched int: %+v", probe)
+	}
+
+	svcPort = v1.ServicePort{
+		Port: 22,
+	}
+	probe = findProbeForPods(pods, &svcPort)
+	if probe != nil {
+		t.Errorf("ServicePort.Port mismatch: %+v", probe)
+	}
+}
+
+func TestGetPodOwnerTypeAndName(t *testing.T) {
+	tests := []struct {
+		desc    string
+		expType string
+		expName string
+		pod     *v1.Pod
+	}{
+		{
+			desc:    "deployment",
+			expType: "deployment",
+			expName: "deploy-name",
+			pod:     &v1.Pod{ObjectMeta: createTestObjMeta("Deployment", "deploy-name", true)},
+		},
+		{
+			desc:    "stateful set",
+			expType: "statefulset",
+			expName: "statefulset-name",
+			pod:     &v1.Pod{ObjectMeta: createTestObjMeta("StatefulSet", "statefulset-name", true)},
+		},
+		{
+			desc:    "daemon set",
+			expType: "daemonset",
+			expName: "daemonset-name",
+			pod:     &v1.Pod{ObjectMeta: createTestObjMeta("DaemonSet", "daemonset-name", true)},
+		},
+		{
+			desc:    "replica set with no pod hash",
+			expType: "deployment",
+			expName: "replicaset-name",
+			pod:     &v1.Pod{ObjectMeta: createTestObjMeta("ReplicaSet", "replicaset-name", false)},
+		},
+		{
+			desc:    "replica set with pod hash",
+			expType: "deployment",
+			expName: "replicaset-name",
+			pod: &v1.Pod{
+				ObjectMeta: createTestObjMeta("ReplicaSet", "replicaset-name-67c6f7c5fd", true),
+			},
+		},
+		{
+			desc:    "nil controller should use default values",
+			expType: "deployment",
+			expName: "deploy-name",
+			pod: &v1.Pod{
+				ObjectMeta: meta_v1.ObjectMeta{
+					OwnerReferences: []meta_v1.OwnerReference{
+						{
+							Name:       "deploy-name",
+							Controller: nil,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.desc, func(t *testing.T) {
+			actualType, actualName := getPodOwnerTypeAndName(test.pod)
+			if actualType != test.expType {
+				t.Errorf("getPodOwnerTypeAndName() returned %s for owner type but expected %s", actualType, test.expType)
```

Comment:
neat picking: the new methods and functions are exported, but they are not exported (start with a lowercase letter) in error messages in the tests

Reply:
I don't follow you I'm afraid, can you explain the issue?

Reply:
my bad! I thought `getPodOwnerTypeAndName()` was `GetPodOwnerTypeAndName()` for some reason :(